src/Microsoft.DotNet.TestFramework/Extensions/DirectoryInfoExtensions.cs

```diff
@@ -21,5 +21,15 @@ public static FileInfo GetFile(this DirectoryInfo subject, string fileName)
         {
             return new FileInfo(Path.Combine(subject.FullName, fileName));
         }
+
+        public static void EnsureExistsEmpty(this DirectoryInfo subject)
```

Comment:
This name could be a bit more natural and clearer, e.g. maybe `EnsureExistsAndEmpty`?